### DISCUSSION TITLE
Tagline, mission, features, goals etc. for Pulsar website

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#50] [General]
Based on the conversation in Discord about getting the news out there, I think there needs to be a discussion on what kind of things we put on the website and show what our goals are so that people don't hear of the project and just find a half done website with a bunch of stuff directly from Atom.

This also very much touches on the topic about [Pulsar's "thing"](https://github.com/orgs/pulsar-edit/discussions/27) discussing Pulsar's USP, what makes it different from everything else and where it can be taken.

## Tagline
~~Currently @kaosine has put in ***Next generation hackable editor for the future*** as the tagline on the website. I have no issues with this but I don't think I've heard any alternatives either so now would be the time to speak up if you feel strongly about it and we can put it to the vote.~~
~~Personally I kind of like the idea of leaning more into the "spacey" nature of it with the language we use, maybe something like "hyper hackable" (not that "Next Generation" doesn't bring to mind some kind of space imagery 😊)~~
~~So some variations I can think of:~~
~~- ***The next generation of the hackable text editor for the future*** - Basically just the above but adds in Pulsar's Atom roots~~
~~- ***Hyper-hackable editor for everyone*** - With the "spacey" hyper- prefix and to imply that it is both for advanced and basic users~~
Chosen as `Community-led Hyper-Hackable Text Editor` by [Discord Poll](https://discord.com/channels/992103415163396136/1040070150298292325/1040070150298292325)

## Mission statement(s)
Now this one is more complicated. Atom's goal was to make a text editor that was a *"zero-compromise combination of hackability and usability:"*. I think Atom fell behind the competition on the former and never really capitalised on the latter so we would need to maybe add more emphasis to this element of it.
Some points I think we could use and expand upon:
- To build upon, not replace, the legacy left behind by Atom and its community
- To design a modern, hyper-hackable editor and ecosystem for everyone regardless of ability
- Extensibility and customisation limited only by your imagination (blegh... cliche and corny I know but you get the idea)
- Community led, developed and focused

## Features
We can obviously just use the ones from Atom but if we do it seems like we should really remix these to make them our own.
Most of the current ones feel a little obvious and out of date, most are things people now expect as the bare minimum so are no longer really "features" to attract people any more.
Instead of focusing on the core features I think it better to focus on what Pulsar is and can be, a description of the sort of software it is rather than a list of things it can do.

So with that said my proposal would be something like this:

**Pulsar**

*The Hyper-hackable text editor for everyone*

- Hyper-hackability
  - Pretty much everything can be hacked, altered and customized in Pulsar. Tweak the look and feel with CSS. Add new features and packages with JavaScript and HTML. Almost nothing is off limits.

- A whole galaxy of packages to explore
  - Pulsar features an extensive array of packages to customise and enhance your editor. Really make Pulsar yours with new themes, languages and almost anything else you can think of.

- Pulsar shines for everybody
  - Pulsar is really designed with all users in mind; from the novice programmer to the seasoned hacker. Out of the box it comes with everything you should need but gives the tools to those who know exactly what they want from their editor to completely alter their experience.

*Our purpose and goals*

- Community developed, led and focused
  - Pulsar is being made by a community who came together from the stellar remnants of Atom. A community that wants to build upon the huge legacy that was left and make a uniquely hackable editor.
- To continue and build upon the legacy of the Atom text editor which is being/has been sunset.
  - This means not only supporting the editor itself but also the package repository with its thousands of community contributions.
- Update core technologies to bring the editor up to date.
  - Things like Node.js, Electron etc. and keeping them up to date so new features and libraries can be used and added without hacky workarounds.
- Emphasize the elements that make the editor great to really make Pulsar stand out from the crowd, not only for ex-Atom users but for everyone.

***

This was probably longer than I originally intended but I'd be interested to hear what others have to say on the topic as I think this is something quite important in order to get people interested in the project and really get it out to the wider community.

**Comment by ThatXliner** [comment]
I think we should mention directly the init script and CSS file, which (in my opinion) is the core of all the customization. It’s like UserScripts and it’s really what made Atom set apart. It puts Atom in the same league as Emacs and Vim but way more accessible because it’s languages everyone should know.

**Comment by confused-Techie** [comment]
So my thoughts on what's being said here, especially since I've just gotten back and haven't had much chance to comment on it as the topics evolved initially. 

## Tagline

I don't have any issue with what @kaosine has written there and is in use on the website, although the alternative I initially came up with when writing our ORG README is "...a true, Community-Based, Hackable, Text Editor".

But on the suggestions you made I personally love the use of "Hyper-Hackable", leaning into the space theme, and where or how that's used I don't feel to strongly. But I also do feel it would be good to mention "Community-Based" especially now with no paid devs working on it, like GitHub had, this is truly a community project and I personally feel it's important to label it as such.

## Mission Statement

I think we should lean into the community aspect. Because it is community led, and focused. But beyond that, like @mauricioszabo said Modern Emacs. We should focus on hackability, then in my opinion making it easier for devs, but I don't think the latter fits into a mission statement.

That is to say a community led, hackable platform. We make it easier and more hackable than before and that's our goal. Provide a modern platform that people can hack on.

## Features

As for those last bullets points you made, I love those. The *Our Purpose and Goals* resonate largely with how I feel myself. And personally I think thats the most honest description of what we are doing, and at least until everything else is ironed out should be our motto for now. I think that covers almost everything we are all working on in an honest way and paints an honest yet complete picture of what we are doing. And really is something we should probably add directly to our readme as well as other places.

**Reply by Daeraxa** [reply]
I definitely agree on the community aspect, I tried to get that in with the "Community developed, led and focused" section (which I can now see is in the wrong section of my post...) but I think leaning even more heavily on it could be really good.

**Reply by confused-Techie** [reply]
Thinking more on taglines, how would we feel about this: "A Community-Led, Next-Generation, Hyper-Hackable Text Editor for the Future".

Is that to much of a mouthful? Thought it sufficiently covered all of our points as a tagline for the website and Org.

**Reply by Daeraxa** [reply]
With something like that I think we would need to pick either "next-generation" *or* "for the future" as I think they really cover the same aspect/goal and make it a little less wordy.

**Reply by mauricioszabo** [reply]
Something like _A community-led hackable text editor for the future_ sounds amazing too, and not too much of a mouthful :)

**Reply by DeeDeeG** [reply]
I like the sound of all these.

With all this tech debt is sometimes feels like we are an editor of the past... :sweat_smile:  But the vision is very much "future".

And "community" I agree with totally, since that is I suppose what differentiates us from most other editors is following in Atom's footsteps of community focus.
